### PR TITLE
Updates to the style guide

### PR DESCRIPTION
**Pull request by odow**
Ref #1780.

### CODE REVIEW

File: docs/src/style.md [hunk 1]
```diff
@@ -167,6 +167,76 @@ empty vector with element type `T`. Prefer `T[]` because it is more concise.
...
+```julia
+@variable(model, x >= 0)
+```
+
+Bad:
+```julia
+@variable(model, x >= 0.0)
+```
+
+Always use parentheses.
```
Comment (blegat) [inline docs/src/style.md]: This is only in the context of macros right ? Otherwise it is ambiguous

File: docs/src/style.md [hunk 2]
```diff
@@ -167,6 +167,76 @@ empty vector with element type `T`. Prefer `T[]` because it is more concise.
 Both `1.0` and `1.` create a `Float64` with value `1.0`. Prefer `1.0` over
 `1.` because it is more easily distinguished from the integer constant `1`.
 
+#### Comments
+
+Comments in code must be proper English sentences with appropriate punctuation.
```
Comment (mlubin) [inline docs/src/style.md]: Provide a motivation, e.g., for nonnative speakers and for general clarity.

File: docs/src/style.md [hunk 3]
```diff
@@ -167,6 +167,76 @@ empty vector with element type `T`. Prefer `T[]` because it is more concise.
...
+```julia
+# This is a comment demonstrating a good comment.
+```
+
+Bad:
+```julia
+# a bad comment
+```
+
+#### Macro syntax
```
Comment (mlubin) [inline docs/src/style.md]: Is this supposed to apply to all macros or just JuMP's DSL?

File: docs/src/style.md [hunk 4]
```diff
@@ -167,6 +167,76 @@ empty vector with element type `T`. Prefer `T[]` because it is more concise.
...
+
+Never use `1.0` when `1` is okay.
```
Comment (mlubin) [inline docs/src/style.md]: Why would the use of 1.0 or 1 depend on if it's inside a macro?

Comment (mlubin) [inline docs/src/style.md]: Motivation?

File: docs/src/style.md [hunk 5]
```diff
@@ -179,6 +249,18 @@ some_local_variable = ...
 some_file.jl # Except for ModuleName.jl.
 ```
 
+Begin private functions and constants with an underscore.
```
Comment (mlubin) [inline docs/src/style.md]: Explain that all other functions are exported (after #1779)

File: docs/src/style.md [hunk 6]
```diff
@@ -167,6 +167,76 @@ empty vector with element type `T`. Prefer `T[]` because it is more concise.
...
+```julia
+@constraint(model, 3 * x <= 1)
+```
+
+Bad:
+```julia
+@constraint(model, x * 3 <= 1)
+```
+
+Prefer plural form of macros over lots of repeated calls to singular forms.
```
Comment (mlubin) [inline docs/src/style.md]: Motivation? Easier to read? What about calls with lots of keyword arguments?

**Pull request closed without merging**